validate/validate.go

```diff
@@ -0,0 +1,163 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/kong/deck/state"
+	"github.com/kong/deck/utils"
+	"github.com/kong/go-kong/kong"
+)
+
+type Validator struct {
+	ctx               context.Context
+	state             *state.KongState
+	client            *kong.Client
+	parallelism       int
+	rbacResourcesOnly bool
+}
+
+func NewValidator(
+	ctx context.Context,
+	ks *state.KongState,
+	client *kong.Client,
+	parallelism int,
+	rbacResourcesOnly bool,
```

Comment:
Instead create a ValidaterOpt struct and then take that struct as input.

```go
type ValidatorOpts struct {
.
.

}

func NewValidator(opt ValidatorOpt) {
.
.
}
 ```

Reply:
Done.